app/models/dimensions/item.rb

```diff
@@ -1,3 +1,57 @@
+require 'ostruct'
+require 'json'
+
 class Dimensions::Item < ApplicationRecord
   validates :content_id, presence: true
+
+  def get_content
+    json_object = JSON.parse(self.raw_json, object_class: OpenStruct)
+    return if json_object.blank?
+    extract_by_schema_type(json_object)
+  end
+
+private
+
+  VALID_SCHEMA_TYPES = %w[
+    answer
+    case_study
+    consultation
+    corporate_information_page
+    detailed_guide
+    document_collection
+    fatality_notice
+    help
+    hmrc_manual_section
+    html_publication
+    manual
+    manual_section
+    news_article
+    publication
+    service_manual_guide
+    simple_smart_answer
+    specialist_document
+    speech
+    statistical_data_set
+    take_part
+    topical_event_about_page
+    working_group
+    world_location_news_article
+  ].freeze
+
+  def extract_by_schema_type(json)
+    schema = json.schema_name
+    return if schema.nil?
```

Comment:
What about following the same approach of raising an exception is Schema is nil? In theory, all content items should have an Schema.